components/datarooms/settings/archive-dataroom.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button } from "@/components/ui/button";
+import {
+  Card,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+
+import useArchiveDataroom from "@/lib/swr/use-archive-dataroom";
+
+export default function ArchiveDataroom({
+  dataroomId,
+  teamId,
+}: {
+  dataroomId: string;
+  teamId?: string;
+}) {
+  const { isArchived, toggleArchive, isLoading, isUpdating } =
+    useArchiveDataroom(dataroomId, teamId);
```

Comment:
Instead of creating this, you can just pass a prop to the component `isArchived` and then add the functionality to this component for archiving / unarchiving. 

While it works with SWR that's adding a bit of confusion.

Reply:
Actually, I did this way because if you see the image,
![image](https://github.com/user-attachments/assets/695a46f3-30cf-449e-9b7d-b1de37f2419b)

This component has it's own functionality, which I added inside it. I created that hook to keep away the business logic from the component so components just renders and we handle the logic in hook. It's not confusing, I've seen useDataroom() hook as well :)